altmetric/explorer/api/filters.py

```diff
@@ -0,0 +1,63 @@
+class Filters:
+    '''Holds a list of filters and provides methods that transform them
+     into query parameters and message digests.'''
+
+    def __init__(self):
+        """Initializes a new Filters object"""
+        self.items = []
+
+    def add_filter(self, arg, value):
+        """Add a new filter to the list of filters
+
+        Args:
+            arg (string): name of the filter
+            value (string, list, tuple or set): filter value(s)
+
+        Returns:
+            Filters: self
+        """
+        self.items.append((arg, value))
+        return self
+
+    def __len__(self):
+        """Count of filters
+
+        Returns:
+            integer: number of filters stored
+        """
+        return len(self.items)
+
+    def message(self):
+        """Convert the Filters to a message that can be used to construct a
+        digest for authentication.
+
+        See https://www.altmetric.com/explorer/documentation/api#authentication
+
+        Returns:
+            string: the digest
+        """
+        result = []
+        for arg, value in sorted(self.items):
```

Comment:
I think you'd usually do this in a comprehension in Python but maybe you've seen something different in your past role(s).  I'm not sure I can see a way to condense this down to a comprehension without adding an extra method to unpack each item in `self.items` which seems a bit much for a simple loop (to me at least).  Have you maybe spotted something I missed though?

nb `reduce` exists but it's hidden away in a library and it's not all that useful because Python's `lambda` does allow multiple line function bodies so it's only useful for simple one-lines.